packages/react-data-grid/src/CellAction.js

```diff
@@ -0,0 +1,82 @@
+import classNames from 'classnames';
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const CellActionShape = require('./PropTypeShapes/CellActionShape');
+
+class CellAction extends React.Component {
+  static propTypes = {
+    action: PropTypes.shape(CellActionShape).isRequired,
+    isFirst: PropTypes.bool.isRequired,
+    cellHeight: PropTypes.number.isRequired
+  }
+
+  constructor(props) {
+    super(props);
+
+    this.state = {isMenuOpen: false};
+  }
+
+  onToggleMenu = () => {
+    this.setState({isMenuOpen: !this.state.isMenuOpen});
```

Comment:
If the next state is dependent on the previous state then the recommended approach is to use the function overload of `setState`
```
this.setState((prevState) => {
  return { isMenuOpen: !prevState.isMenuOpen };
});
```
https://medium.com/@shopsifter/using-a-function-in-setstate-instead-of-an-object-1f5cfd6e55d1